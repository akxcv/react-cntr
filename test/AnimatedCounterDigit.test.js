import { shallowToJson } from 'enzyme-to-json'
import { AnimatedCounterDigitBuilder as Builder } from './support/builders'

it('matches snapshot', function () {
  const component = Builder.shallow()
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})

it('limits the value to maxValue', function () {
  const component = Builder.shallow({ maxValue: 5 })
  expect(component.find('.rollex-digit-lane-label').length).toBe(7)
})

it('sets correct initial zero position', function () {
  const downComponent = Builder.shallow({ direction: 'down', height: 1 })
  expect(downComponent.state().initialZeroPosition).toBe(-10)
  const downComponentWithMax = Builder.shallow({ direction: 'down', maxValue: 2, height: 1 })
  expect(downComponentWithMax.state().initialZeroPosition).toBe(-10)

  const upComponent = Builder.shallow({ direction: 'up', height: 1 })
  expect(upComponent.state().initialZeroPosition).toBe(0)
  const upComponentWithMax = Builder.shallow({ direction: 'up', maxValue: 2, height: 1 })
  expect(upComponentWithMax.state().initialZeroPosition).toBe(-7)
})

it('sets correct final zero position', function () {
  const downComponent = Builder.shallow({ direction: 'down', height: 1 })
  expect(downComponent.state().finalZeroPosition).toBe(0)
  const downComponentWithMax = Builder.shallow({ direction: 'down', maxValue: 2, height: 1 })
  expect(downComponentWithMax.state().finalZeroPosition).toBe(-7)

  const upComponent = Builder.shallow({ direction: 'up', height: 1 })
  expect(upComponent.state().finalZeroPosition).toBe(-10)
  const upComponentWithMax = Builder.shallow({ direction: 'up', maxValue: 2, height: 1 })
  expect(upComponentWithMax.state().finalZeroPosition).toBe(-10)
})

test('reset() is being called on transition end', function () {
  const component = Builder.mount()
  component.instance().reset = jest.genMockFunction()
  component.instance().forceUpdate() // required for the mock to work
  component.find('.rollex-digit').simulate('transitionEnd')
  expect(component.instance().reset.mock.calls.length).toBe(1)
})
