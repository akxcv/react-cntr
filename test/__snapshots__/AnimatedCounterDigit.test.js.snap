// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="rollex-digit"
  onTransitionEnd={[Function]}
  style={
    Object {
      "transform": "translateY(0px)",
      "transitionDuration": "300ms",
      "transitionProperty": "transform",
      "transitionTimingFunction": "ease-in",
      "willChange": "transform",
    }
  }
>
  <div
    className="rollex-digit-lane-label"
    style={
      Object {
        "height": 18,
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    className="rollex-digit-lane-label"
    style={
      Object {
        "height": 18,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="rollex-digit-lane-label"
    style={
      Object {
        "height": 18,
      }
    }
  >
    <div>
      2
    </div>
  </div>
  <div
    className="rollex-digit-lane-label"
    style={
      Object {
        "height": 18,
      }
    }
  >
    <div>
      3
    </div>
  </div>
  <div
    className="rollex-digit-lane-label"
    style={
      Object {
        "height": 18,
      }
    }
  >
    <div>
      4
    </div>
  </div>
  <div
    className="rollex-digit-lane-label"
    style={
      Object {
        "height": 18,
      }
    }
  >
    <div>
      5
    </div>
  </div>
  <div
    className="rollex-digit-lane-label"
    style={
      Object {
        "height": 18,
      }
    }
  >
    <div>
      6
    </div>
  </div>
  <div
    className="rollex-digit-lane-label"
    style={
      Object {
        "height": 18,
      }
    }
  >
    <div>
      7
    </div>
  </div>
  <div
    className="rollex-digit-lane-label"
    style={
      Object {
        "height": 18,
      }
    }
  >
    <div>
      8
    </div>
  </div>
  <div
    className="rollex-digit-lane-label"
    style={
      Object {
        "height": 18,
      }
    }
  >
    <div>
      9
    </div>
  </div>
  <div
    className="rollex-digit-lane-label"
    style={
      Object {
        "height": 18,
      }
    }
  >
    <div>
      0
    </div>
  </div>
</div>
`;
